documentation/language_server.tex

```diff
@@ -228,34 +226,40 @@ \section{Example: hover request handling}
 \sectionSrc{Request Manager}
 {language\_server/src/request\_manager.h}
 
-\SC{Request manager} encapsulates a queue of requests with a worker thread that processes them. There may be up to two \SC{dispatcher} instances in the language server: one for LSP and one for DAP. Both of them add the requests they parse into one \SC{request manager}. It is necessary to process the requests one by one, because the parser library cannot process more requests at the same time.
+\TT{Request manager} encapsulates a queue of requests with a worker thread that processes them. There may be up to two \TT{dispatcher} instances in the language server: one for LSP and one for DAP. Both of them add the requests they parse into one \TT{request manager}. It is necessary to process the requests one by one, because the parser library cannot process more requests at the same time.
 
 There are three threads running in the language server:
 \begin{itemize}
-	\item LSP read thread --- a thread in which the \TT{dispatcher} reads messages from a standard input.
+	\item LSP read thread --- a thread in which the \TT{dispatcher} reads messages from the standard input.
 	\item DAP read thread --- a thread in which the \TT{tcp\_handler} listens on a localhost port to initiate a DAP session. After accepting the DAP client, the \TT{dispatcher} reads DAP input on this thread too.
 	\item Worker thread in \TT{request\_manager} that processes each request using the \TT{lsp\_server} or the \TT{dap\_server} and ultimately the parser library.
 \end{itemize}
 
-The threads are synchronized in two ways: firstly, there is a mutex that protects adding to the request queue simultaneously by the LSP and the DAP read threads. Secondly, there is a conditional variable to control the worker thread so it sleeps when there are no requests and wakes up when a new request has been added.
+The threads are synchronized in two ways: First, there is a mutex that protects adding to the request queue simultaneously by the LSP and the DAP read threads. Second, there is a conditional variable to control the worker thread.
 
-In the request manager, there is a mechanism for invalidating requests that have been obsoleted by new requests. When a new request comes, all previous requests (including the currently processed one) that are concerning the same file are invalidated, but they cannot be simply removed from the queue and they still have to be processed. However, the parser library gets the information that the request has been obsoleted by a newer one and may behave differently.
+Request manager additionally incorporates a mechanism for invalidating requests that have been obsoleted by new requests. The obsoleting of requests is done by a cancellation token. It is shared between the parser library and the \TT{request\_manager}. When set to true, the results of current request or notification are no longer needed, the parser library stops all parsing and return as soon as possible.
```

Comment:
```suggestion
Request manager additionally incorporates a mechanism for invalidating requests that are being obsoleted by new requests. The obsoleting of requests is done by a cancellation token. It is shared between the parser library and the \TT{request\_manager}. When set to true, the results of current request or notification are no longer needed, the parser library stops all parsing and return as soon as possible.
```

Reply:
tu by som nechal ten perfect, tie veci boli obsoleted v momente ked uzivatel nieco napisal, to je minulost